### PR TITLE
Fix payload size limit

### LINKED ISSUE #223: Payload too large error
I don’t get any error related to the payload size limit.

- [ ] The default is 100Mb, but when I push a dataset of 200Mb, I don’t get any error.

I should get a [`413 Payload Too Large`](https://developer.mozilla.org/fr/docs/Web/HTTP/Status/413) with the following JSON body

```json
{
  "message": "Payload too large",
  "errorCode": "payload_too_large",
  "errorType": "invalid_request_error",
  "errorLink": "https://docs.meilisearch.com/errors#payload_too_large"
}
```

- [ ] If I set the `--http-payload-size-limit=1`, I don’t get any error when pushing a 10Mb dataset. But again, I should have the same behavior as described in the previous point.

### PR DESCRIPTION
**Pull request by MarinPostma**
Fix #223

This was due to the fact that Payload ignores the limit payload size limit. I fixed it by implementing my own `Payload` extractor that checks that the size of the payload is not too large.

I also refactored the `create_app` a bit.

### CODE REVIEW

**Review by curquiza — APPROVED** [review]
👌

File: meilisearch-http/src/index_controller/update_actor/actor.rs [hunk 1]
```diff
@@ -137,21 +137,38 @@ where
...
-                    Some((file, update_file_id))
-                } else {
-                    // empty update, delete the empty file.
-                    fs::remove_file(&path).await?;
-                    None
+                let file_len = write_to_file(&mut file, payload).await;
+
+                match file_len {
+                    Ok(len) if len > 0 => {
+                        file.flush().await?;
```
Comment (irevoire) [inline meilisearch-http/src/index_controller/update_actor/actor.rs:160]: if the flush fail shouldn't we delete the half-complete file?

Comment (MarinPostma) [reply meilisearch-http/src/index_controller/update_actor/actor.rs:160]: that's right, I'll move this in the `write_to_file` function.

**Review by irevoire — APPROVED** [review]
The code looks good to me :slightly_smiling_face:

**Review by curquiza — APPROVED** [review]

**Comment by MarinPostma** [comment]
bors merge

**Pull request merged**